### DISCUSSION TITLE
Idle events

### DISCUSSION THREAD

**Discussion by skullicious** [giorgiabosello/google-maps-react-markers#148] [Help]
Hi guys,

I'm trying to get at the idle event so I can hook into it at some level from Cypress to test the map once it's finished zooming and panning (snapshot image diffs are slightly off sometimes, so I'm having to add wait(10000) to all of my assertions.

I tried natively adding with 

`useEffect(() => {
    if (mapReady) {
        const listener = window.google.maps.event.addListener(mapRef.current, 'idle', handleIdle);

        return () => {
            `window.google.maps.event.removeListener(listener);`
        };
    }
}, [mapReady]);`

But it only fires once when I first load the page.

I had a look at the Events array prop but that only seems to support standard DOM events.

So I thought I'd try IdleContent prop and just use the presence of that in the markup but I don't see that rendered anywhere in the markup (I expected it would be rendered the same as LoadingContent).

Any ideas?

Thanks!

**Comment by giorgiabosello** [comment]
@skullicious Hi, I pushed a new version a few hours before which resolved an issue with the idle events. Would you try if your event is working now?

**Reply by skullicious** [reply]
Hi Giorgia, I haven't had a chance to test this yet as I had to roll back to version 1.4.2 since the newer version was breaking some react tests.

I tried a few workarounds but didn't get anywhere..

Shall I raise another issue for this?

![error](https://github.com/giorgiabosello/google-maps-react-markers/assets/38982429/d530507c-e595-451d-af25-28f4c4be2b4b)

**Reply by giorgiabosello** [reply]
@skullicious it doesn't seem to be a problem of the library, imports haven't changed.
See if this helps: https://stackoverflow.com/questions/58211880/uncaught-syntaxerror-cannot-use-import-statement-outside-a-module-when-import, maybe you are missing the `type: "module"` in your package.json?
